rocketpy/Flight.py

```diff
@@ -1563,6 +1575,258 @@ def uDot(self, t, u, postProcessing=False):
+            * self.rocket._csys
+        )
+        S_noz_33 = 0.5 * self.rocket.motor.nozzleRadius**2
+        S_noz_11 = 0.5 * S_noz_33 + 0.25 * r_NOZ**2
+        S_noz_22 = S_noz_11
+        S_noz_12 = 0
+        S_noz_13 = 0
+        S_noz_23 = 0
+        S_nozzle = Matrix(
+            [
+                [S_noz_11, S_noz_12, S_noz_13],
+                [S_noz_12, S_noz_22, S_noz_23],
+                [S_noz_13, S_noz_23, S_noz_33],
+            ]
+        )
+        ## Inertia tensor
+        I_11 = self.rocket.I_11.getValueOpt(t)
+        I_12 = self.rocket.I_12.getValueOpt(t)
+        I_13 = self.rocket.I_13.getValueOpt(t)
+        I_22 = self.rocket.I_22.getValueOpt(t)
+        I_23 = self.rocket.I_23.getValueOpt(t)
+        I_33 = self.rocket.I_33.getValueOpt(t)
+        I = Matrix(
+            [
+                [I_11, I_12, I_13],
+                [I_12, I_22, I_23],
+                [I_13, I_23, I_33],
+            ]
+        )
+        ## Inertia tensor time derivative in the body frame
+        I_11_dot = self.rocket.I_11.differentiate(t)
+        I_12_dot = self.rocket.I_12.differentiate(t)
+        I_13_dot = self.rocket.I_13.differentiate(t)
+        I_22_dot = self.rocket.I_22.differentiate(t)
+        I_23_dot = self.rocket.I_23.differentiate(t)
+        I_33_dot = self.rocket.I_33.differentiate(t)
+        I_dot = Matrix(
+            [
+                [I_11_dot, I_12_dot, I_13_dot],
+                [I_12_dot, I_22_dot, I_23_dot],
+                [I_13_dot, I_23_dot, I_33_dot],
+            ]
+        )
+        ## Inertia tensor relative to CM
+        H = (r_CM.cross_matrix @ -r_CM.cross_matrix) * total_mass
+        I_CM = I - H
+
+        # Prepare transformation matrices
+        K = Matrix.transformation(e)
+        Kt = K.transpose
+
+        # Compute aerodynamic forces and moments
+        R1, R2, R3, M1, M2, M3 = 0, 0, 0, 0, 0, 0
+
+        ## Drag force
+        rho = self.env.density.getValueOpt(z)
+        windVelocityX = self.env.windVelocityX.getValueOpt(z)
+        windVelocityY = self.env.windVelocityY.getValueOpt(z)
+        wind_velocity = Vector([windVelocityX, windVelocityY, 0])
+        freestreamSpeed = abs((wind_velocity - Vector(v)))
+        freestreamMach = freestreamSpeed / self.env.speedOfSound.getValueOpt(z)
+        if t < self.rocket.motor.burnOutTime:
+            dragCoeff = self.rocket.powerOnDrag.getValueOpt(freestreamMach)
+        else:
+            dragCoeff = self.rocket.powerOffDrag.getValueOpt(freestreamMach)
+        R3 += -0.5 * rho * (freestreamSpeed**2) * self.rocket.area * (dragCoeff)
+
+        ## Off center moment
+        M1 += self.rocket.cpEccentricityY * R3
+        M2 -= self.rocket.cpEccentricityX * R3
+
+        # Get rocket velocity in body frame
+        vB = Kt @ v
+        # Calculate lift and moment for each component of the rocket
+        for aeroSurface, position in self.rocket.aerodynamicSurfaces:
```

Comment:
This loop is being repeated in the different uDot functions. 
Maybe a generic function should be created to perform this loop separatly?

I know it could slow down the code a little bit, but it would also keep everything more organized in this file.

Reply:
A `calculateAerodynamicExternalReactions` function could solve this. I doubt that will slow anything down